cli/src/commands/split.rs

```diff
@@ -179,16 +174,33 @@ the operation will be aborted.
     // commit.
     tx.mut_repo()
         .set_rewritten_commit(commit.id().clone(), second_commit.id().clone());
-    let num_rebased = if args.siblings {
-        rebase_children_for_siblings_split(
-            &mut tx,
-            command.settings(),
-            &commit,
-            vec![first_commit.clone(), second_commit.clone()],
-        )?
-    } else {
-        tx.mut_repo().rebase_descendants(command.settings())?
-    };
+    let mut num_rebased = 0;
+    tx.mut_repo().transform_descendants(
+        command.settings(),
+        vec![commit.id().clone()],
+        |mut rewriter| {
+            let old_parents = rewriter.old_commit().parent_ids();
+            // Reserve space up front since we know we're adding at most one more parent.
+            let mut new_parents = Vec::with_capacity(old_parents.len() + 1);
+            for parent in old_parents {
```

Comment:
We should probably add some more helper to `CommitRewriter` to make it easier to do this kind of transformation. We can do that later.

Reply:
Ack.

Reply:
Actually, the existing `replace_parent()` should already work well here. I think this should work:

```rust
if args.siblings {
    rewriter.replace_parent(second_commit.id(), &[first_commit.id().clone(), second_commit.id().clone()]);
}
```

Reply:
Yep, this works. We don't need to do anything for the non-siblings case since we already marked the original commit as rewritten.